src/lib/y2network/dialogs/wireless_networks.rb

```diff
@@ -0,0 +1,137 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast2/feedback"
+require "y2network/widgets/wireless_networks"
+require "y2network/wireless_network"
+require "cwm/popup"
+
+Yast.import "Label"
+
+module Y2Network
+  module Dialogs
+    # Button that runs an custom action when it is clicked
+    class CallbackButton < ::CWM::PushButton
+      # @param label [String] Button label
+      # @param block [Proc] Action to run
+      def initialize(label, &block)
+        @label = label
+        @block = block
+      end
+
+      # @see CWM::AbstractWidget
+      def label
+        @label
+      end
+
+      # @see CWM::CustomWidget
+      def handle
+        @block.call
+
+        nil
+      end
+    end
+
+    # This widget displays a list of wireless networks and allows the user to select one
+    #
+    # @example Returning the ESSID of the selected network
+    #   WirelessNetworks.new("wlo1").run #=> "sample_essid"
+    class WirelessNetworks < CWM::Popup
+      attr_reader :interface
+
+      # Constructor
+      #
+      # @param builder [InterfaceConfigBuilder]
+      def initialize(builder)
+        textdomain "network"
+        @builder = builder
+      end
+
+      # @see CWM::AbstractWidget
+      def title
+        _("Available Wireless Networks")
```

Comment:
Wireless could be redundant in this context, we already removed from the frames, so maybe we could omit it here too, but nothing important to be addressed.